client/my-sites/site-settings/date-time-format/date-format-option.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import { phpToMomentDatetimeFormat } from 'lib/formatting';
+import { defaultDateFormats } from './default-formats';
+
+export const DateFormatOption = ( {
+	dateFormat,
+	disabled,
+	isCustom,
+	localizedDate,
+	setCustomDateFormat,
+	setDateFormat,
+	translate,
+} ) => (
+	<FormFieldset>
+		<FormLabel>
+			{ translate( 'Date Format' ) }
+		</FormLabel>
+		{ defaultDateFormats.map( ( format, index ) =>
+			<FormLabel key={ index }>
+				<FormRadio
+					checked={ ! isCustom && format === dateFormat }
+					disabled={ disabled }
+					name="date_format"
+					onChange={ setDateFormat }
+					value={ format }
+				/>
+				<span>{ localizedDate.format( phpToMomentDatetimeFormat( format ) ) }</span>
+			</FormLabel>
+		) }
+		<FormLabel className="date-time-format__custom-field">
+			<FormRadio
+				checked={ isCustom }
+				disabled={ disabled }
+				name="date_format"
+				onChange={ setCustomDateFormat }
+				value={ dateFormat }
+			/>
+			<span>
+				{ translate( 'Custom', { comment: 'Custom date/time format field' } ) }
+				<FormInput
+					disabled={ disabled }
+					name="date_format_custom"
+					onChange={ setCustomDateFormat }
+					type="text"
+					value={ dateFormat || '' }
+				/>
+				<FormSettingExplanation>
+					{ isCustom && dateFormat
+						? translate( 'Preview: %s', {
+							args: localizedDate.format( phpToMomentDatetimeFormat( dateFormat ) ),
+							comment: 'Date/time format preview',
+						} )
+						: ''
```

Comment:
I'm confused at the ternary here. Is the goal only to output a string when `isCustom && dateFormat`? If so, we can just just another `&&`

```js
{ isCustom && dateFormat && translate(
	'Preview: %s',
	{
		args: …,
		comment: …,
	}
) }
```